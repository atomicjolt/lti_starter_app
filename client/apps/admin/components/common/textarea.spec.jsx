import React from 'react';
import TestRenderer from 'react-test-renderer';
import Textarea from './textarea';

describe('textarea', () => {
  let result;
  let props;
  let changed;
  let instance;

  beforeEach(() => {
    changed = false;
    props = {
      textareaProps: {
        id: 'IM AN ID',
        value: 'IM A VALUE',
        disabled: false,
        name: 'the name',
        placeholder: 'IMA PLACEHOLDER',
        maxLength: 1,
        minLength: 1,
        cols: 3,
        rows: 2,
        onChange: () => { changed = true; },
      },
      className: 'imaclass',
      labelText: 'IMA LABEL',
    };
    result = TestRenderer.create(<Textarea {...props} />);
    instance = result.root;
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the onChange event', () => {
    expect(changed).toBeFalsy();
    instance.findByType('textarea').simulate('change');
    expect(changed).toBeTruthy();
  });
});
