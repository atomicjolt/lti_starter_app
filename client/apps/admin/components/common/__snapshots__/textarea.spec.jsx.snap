// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Textarea
    className="imaclass"
    labelText="IMA LABEL"
    textareaProps={
      Object {
        "cols": 3,
        "disabled": false,
        "id": "IM AN ID",
        "maxLength": 1,
        "minLength": 1,
        "name": "the name",
        "onChange": [Function],
        "placeholder": "IMA PLACEHOLDER",
        "rows": 2,
        "value": "IM A VALUE",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          IMA LABEL
        </span>,
        <textarea
          cols={3}
          disabled={false}
          id="IM AN ID"
          maxLength={1}
          minLength={1}
          name="the name"
          onChange={[Function]}
          placeholder="IMA PLACEHOLDER"
          rows={2}
          value="IM A VALUE"
        />,
        undefined,
      ],
      "className": "imaclass",
      "htmlFor": "IM AN ID",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "IMA LABEL",
        },
        "ref": null,
        "rendered": "IMA LABEL",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cols": 3,
          "disabled": false,
          "id": "IM AN ID",
          "maxLength": 1,
          "minLength": 1,
          "name": "the name",
          "onChange": [Function],
          "placeholder": "IMA PLACEHOLDER",
          "rows": 2,
          "value": "IM A VALUE",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      undefined,
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            IMA LABEL
          </span>,
          <textarea
            cols={3}
            disabled={false}
            id="IM AN ID"
            maxLength={1}
            minLength={1}
            name="the name"
            onChange={[Function]}
            placeholder="IMA PLACEHOLDER"
            rows={2}
            value="IM A VALUE"
          />,
          undefined,
        ],
        "className": "imaclass",
        "htmlFor": "IM AN ID",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "IMA LABEL",
          },
          "ref": null,
          "rendered": "IMA LABEL",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cols": 3,
            "disabled": false,
            "id": "IM AN ID",
            "maxLength": 1,
            "minLength": 1,
            "name": "the name",
            "onChange": [Function],
            "placeholder": "IMA PLACEHOLDER",
            "rows": 2,
            "value": "IM A VALUE",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        undefined,
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
